### PR TITLE
PT-2936 Collection search by name

### PR DESCRIPTION
**Pull request by sanufriev**

### CODE REVIEW

File: api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/CollectionElasticService.kt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package com.rarible.protocol.union.api.service.elastic
 
+import co.elastic.clients.elasticsearch.ml.Page
```
Comment (dmitry-saraykin) [inline api/src/main/kotlin/com/rarible/protocol/union/api/service/elastic/CollectionElasticService.kt:3]: unused

File: core/src/main/resources/mappings/collection.json [hunk 2]
```diff
@@ -36,7 +94,65 @@
     "meta": {
       "properties": {
```
Comment (dmitry-saraykin) [inline core/src/main/resources/mappings/collection.json:95]: I think "description" is missing

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt [hunk 3]
```diff
@@ -11,14 +12,37 @@ import org.springframework.stereotype.Component
...
+            "name.initials" to boost(100),
+            "name.keyword" to boost(100),
+            "name.specialCharacter" to boost(100),
+
+            "meta.name" to boost(100),
+            "meta.name.camelCase" to boost(100),
+            "meta.name.join" to boost(100),
+            "meta.name.initials" to boost(100),
+            "meta.name.keyword" to boost(100),
+            "meta.name.specialCharacter" to boost(100),
```
Comment (dmitry-saraykin) [inline enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt:32]: also here description is missing

File: core/src/main/kotlin/com/rarible/protocol/union/core/model/elastic/EsCollection.kt [hunk 4]
```diff
@@ -39,6 +39,7 @@ data class EsCollection(
 
     data class CollectionMeta(
         val name: String,
+        val description: String? = null,
```
Comment (dmitry-saraykin) [inline core/src/main/kotlin/com/rarible/protocol/union/core/model/elastic/EsCollection.kt:42]: this is not assigned anywhere. Either remove it or add full support in mappings and queries

File: enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt (hunk 3 above)
Comment (sanufriev) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt:32]: Do you think it needed? MP not use for search

Comment (dmitry-saraykin) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt:32]: on mp we found it not useful. I just noticed that you have in CollectionMeta but don't have it anywhere else. Need to make it consistent

Comment (sanufriev) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt:32]: Ok, then let me just remove it for now

Comment (sanufriev) [reply enrichment/src/main/kotlin/com/rarible/protocol/union/enrichment/repository/search/internal/EsCollectionQueryBuilderService.kt:32]: Done

**Review by dmitry-saraykin — APPROVED** [review]

**Pull request merged**